File: features/fixtures/shared/utils/FixtureConfig.m
//
//  FixtureConfig.m
//  macOSTestApp
//
//  Created by Karl Stenerud on 26.09.24.
//  Copyright © 2024 Bugsnag Inc. All rights reserved.
//

#import "FixtureConfig.h"

@implementation FixtureConfig

- (instancetype) initWithApiKey:(NSString *)apiKey mazeRunnerBaseAddress:(NSURL *)mazeRunnerBaseAddress {
    if ((self = [super init])) {
        _apiKey = apiKey;
        _mazeRunnerURL = mazeRunnerBaseAddress;
        _docsURL = [mazeRunnerBaseAddress URLByAppendingPathComponent:@"docs"];
        _tracesURL = [mazeRunnerBaseAddress URLByAppendingPathComponent:@"traces"];
        _commandURL = [mazeRunnerBaseAddress URLByAppendingPathComponent:@"idem-command"];
        _metricsURL = [mazeRunnerBaseAddress URLByAppendingPathComponent:@"metrics"];
        _notifyURL = [mazeRunnerBaseAddress URLByAppendingPathComponent:@"notify"];
        _sessionsURL = [mazeRunnerBaseAddress URLByAppendingPathComponent:@"sessions"];
        _reflectURL = [mazeRunnerBaseAddress URLByAppendingPathComponent:@"reflect"];
        // configurationURL should be nil by default and only set explicitly in RemoteConfig scenarios
        _configurationURL = nil;
    }
    return self;
}

@end

File: features/fixtures/shared/utils/FixtureConfig.swift
//
//  FixtureConfig.swift
//  iOSTestApp
//
//  Created by Karl Stenerud on 11.03.24.
//  Copyright © 2024 Bugsnag. All rights reserved.
//

import Foundation

@objcMembers class FixtureConfig: NSObject {
    var apiKey: String
    let mazeRunnerURL: URL
    let docsURL: URL
    let tracesURL: URL
    let commandURL: URL
    let metricsURL: URL
    let reflectURL: URL
    var notifyURL: URL
    var sessionsURL: URL

    init(apiKey: String, mazeRunnerBaseAddress: URL) {
        self.apiKey = apiKey
        mazeRunnerURL = mazeRunnerBaseAddress
        docsURL = mazeRunnerBaseAddress.appendingPathComponent("docs")
        tracesURL = mazeRunnerBaseAddress.appendingPathComponent("traces")
        commandURL = mazeRunnerBaseAddress.appendingPathComponent("command")
        metricsURL = mazeRunnerBaseAddress.appendingPathComponent("metrics")
        notifyURL = mazeRunnerBaseAddress.appendingPathComponent("notify")
        sessionsURL = mazeRunnerBaseAddress.appendingPathComponent("sessions")
        reflectURL = mazeRunnerBaseAddress.appendingPathComponent("reflect")
    }
}
